packages/runtime/test-runtime-utils/src/insecureUrlResolver.ts

```diff
@@ -55,15 +55,15 @@ export class InsecureUrlResolver implements IUrlResolver {
             if (!fileName) {
                 throw new Error("FileName should be there!!");
             }
-            return this.resolveHelper(fileName);
+            return this.resolveHelper(fileName, "");
         }
         const parsedUrl = new URL(request.url);
-
         // If hosts match then we use the local tenant information. Otherwise we make a REST call out to the hosting
         // service using our bearer token.
         if (parsedUrl.host === window.location.host) {
             const documentId = parsedUrl.pathname.substr(1).split("/")[0];
-            return this.resolveHelper(documentId);
+            const documentRelativePath = parsedUrl.pathname.replace(`/${documentId}/`,"");
```

Comment:
This does not look completely right. There might not be a "/" after the documentId. The url could just be "https://localhost/document1" or something similar. In that case, the documentRelativePath would be incorrect (You should have a test case for this).

I would do something similar to what you did in LocalResolver.
Get the fullPath as follows and send that to the resolveHelper as the uri instead of sending two params:
`const fullPath = parsedUrl.pathname.substr(1);`

In the createNew case above, the fullPath is the "fileName".

You can then get the documentId in resolveHelper from the uri / fullPath simialr to LocalResolver.

This will further simplify the resolveHelper function because you won't need the if-else clause. You can just append the "fullPath" to the final URL:
fluid://${new URL(this.ordererUrl).host}/${encodedTenantId}/${fullPath}

Reply:
i don't think just passing one parameter will work with the createNew case. Because I can't use split("/") on the `filename` if it doesn't have any "/". Or I need to add a if-else clause to get documentId.

Reply:
also I will have to encode the documentId in fullPath in the final URL

Reply:
Okay. Then its fine to pass in the documentId and the part of the url after the document id. However, the following is still incorrect since there may not be a "/" after the documentId:
`const documentRelativePath = parsedUrl.pathname.replace(`/${documentId}/`,"");`

You should remove the trailing "/" after "documentId" in the replace call.

Reply:
I tested this and for a URL like "http://localhost/documentUrl", documentRelativePath is evaluated incorrectly.

Reply:
if i just remove the "/" after documentId in the replace call, i will still have to handle the "/" if there IS something after documentId in the url. So i added a if clause, please let me know if it can be improved.

Reply:
Yeah, if there is something after the "/" or if the url is "http://localhost/documentUrl/", then you need to handle the extra "/". You can easily do that in the "resolveHelper" by not appending a "/" when adding the documentRelativePath:
``` documentUrl += documentRelativePath`; ```
Actually, that made me realize that in resolveHelper you don't need to check if documentRelativePath is empty, you can always add it to the documentUrl. I will leave a comment there for this as well :-).

I think what you have done works but it feels weird. That "if" condition is unnecessary and is slightly wrong (although the final result is right):
If the url is something like "https://localhost/documentUrl/", the "if" condition is false since the parsedUrl will be "documentUrl/" whereas documentId is "documentUrl". So, you end up trying to get the documentRelativePath which does not seem like the right thing to do (even though it works).

Also, I am not a fan of using "replace". I would rather use "slice". So, the code would look something like:
```
const fullPath = parsedUrl.pathname.substr(1);
const documentId = fullPath.split("/")[0];
const documentRelativePath = fullPath.slice(documentId.length + 1);
```

I will leave it up to you to decide which one of these to use :-).